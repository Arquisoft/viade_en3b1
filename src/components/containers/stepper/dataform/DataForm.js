import React, { Component } from 'react';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';
import DateFnsUtils from '@date-io/date-fns';
import FormHelperText from '@material-ui/core/FormHelperText';
import { MuiPickersUtilsProvider, KeyboardDatePicker } from '@material-ui/pickers';
import CloudUploadIcon from '@material-ui/icons/CloudUpload';
import AddAPhotoIcon from '@material-ui/icons/AddAPhoto';

export class DataForm extends Component {

    handleDateChange = (date) => {
        this.props.handleDateChange(date);
    };

    next = (e) => {
        e.preventDefault();
        this.props.handleNext();
    }

    handleMediaChange = (selectorFiles: FileList, code) => {
        this.props.handleMediaChange(selectorFiles, code);
    }

    render() {

        const { values, handleChange } = this.props;

        return (
            <React.Fragment>
                <Typography variant="h6" gutterBottom>
                    Please, introduce the following data for your new route
                </Typography>
                <form onSubmit={this.next}>
                    <Grid container spacing={3}>
                        <Grid item xs={12} sm={6} style={{ marginTop: 16 }}>
                            <TextField
                                required
                                id="name"
                                name="name"
                                label="Name"
                                fullWidth
                                defaultValue={values.name}
                                autoComplete="routename"
                                onChange={handleChange('name')}
                            />
                        </Grid>
                        <Grid item xs={12} sm={6}>
                            <MuiPickersUtilsProvider utils={DateFnsUtils}>
                                <KeyboardDatePicker
                                    disableToolbar
                                    required
                                    variant="inline"
                                    id="date"
                                    name="date"
                                    format="MM/dd/yyyy"
                                    margin="normal"
                                    label="Date"
                                    value={values.date}
                                    onChange={this.handleDateChange}
                                    KeyboardButtonProps={{
                                        'aria-label': 'date',
                                    }}
                                />
                            </MuiPickersUtilsProvider>
                        </Grid>
                        <Grid item xs={12}>
                            <TextField
                                required
                                id="description"
                                name="description"
                                label="Description"
                                multiline
                                fullWidth
                                defaultValue={values.description}
                                rows="4"
                                variant="outlined"
                                onChange={handleChange('description')}
                            />
                        </Grid>
                        <Grid item xs={12} sm={3} >
                            <input
                                disabled
                                accept="image/*"
                                style={{ display: 'none' }}
                                id="media-photo"
                                multiple
                                defaultValue={values.photos}
                                type="file"
                                onChange={(e) => this.handleMediaChange(e.target.files, 0)} //photos
                            />
                            <label htmlFor="media-photo">
                                <Button disabled variant="contained" color="default" component="span" startIcon={<AddAPhotoIcon />}>
                                    Photos
                            </Button>
                            </label>
                        </Grid>
                        <Grid item xs={12} sm={6} >
                            <input
                                disabled
                                accept="video/*"
                                style={{ display: 'none' }}
                                id="media-video"
                                multiple
                                defaultValue={values.videos}
                                type="file"
                                onChange={(e) => this.handleMediaChange(e.target.files, 1)} //videos
                            />
                            <label htmlFor="media-video">
                                <Button disabled variant="contained" color="default" component="span" startIcon={<CloudUploadIcon />}>
                                    Videos
                            </Button>
                            </label>
                        </Grid>
                        <Grid item xs={12}>
                            <FormHelperText id="my-helper-text">Fields marked with (*) are mandatory.</FormHelperText>
                        </Grid>

                        <Button
                            variant="contained"
                            color="primary"
                            style={{
                                marginTop: 12,
                                marginLeft: 'auto'
                            }}
                            type="submit"
                        >
                            Next
                        </Button>
                    </Grid>
                </form>
            </React.Fragment>
        );
    }
}

export default DataForm;
