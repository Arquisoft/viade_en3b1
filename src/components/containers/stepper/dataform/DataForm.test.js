import DataForm from './DataForm';

const state = {
  activeStep: 0,
  name: '',
  description: '',
  date: new Date(),
  photos: [],
  videos: [],
  points: []
}

const { activeStep, name, description, date, photos, videos, points } = state;
const values = { activeStep, name, description, date, photos, videos, points };

const handleNext = jest.fn();
const handleBack = jest.fn();
const handleChange = jest.fn();
const handleDateChange = jest.fn();
const handleMediaChange = jest.fn();
const handleMapPoints = jest.fn();
const handleDownload = jest.fn();
const createRoute = jest.fn();

test('Define DataForm', () => {
  expect(DataForm).toBeDefined();
});

test('Renders correctly', () => {
  // console.log
  const wrapper = render(<DataForm
    handleNext={handleNext}
    handleChange={handleChange}
    handleDateChange={handleDateChange}
    handleMediaChange={handleMediaChange}
    values={values}
  />);
  expect(wrapper).toMatchSnapshot();
});