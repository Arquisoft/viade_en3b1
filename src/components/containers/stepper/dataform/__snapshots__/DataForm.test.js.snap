// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
Array [
  <h6
    class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
  >
    Please, introduce the following data for your new route
  </h6>,
  <form>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
        style="margin-top:16px"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
            data-shrink="false"
            for="name"
            id="name-label"
          >
            Name
            <span
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-invalid="false"
              autocomplete="routename"
              class="MuiInputBase-input MuiInput-input"
              id="name"
              name="name"
              required=""
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled Mui-required Mui-required"
            data-shrink="true"
            for="date"
            id="date-label"
          >
            Date
            <span
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
              id="date"
              name="date"
              required=""
              type="text"
              value="04/29/2020"
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
            >
              <button
                aria-label="date"
                class="MuiButtonBase-root MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                    />
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink="false"
            for="description"
            id="description-label"
          >
            Description
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline"
          >
            <textarea
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
              id="description"
              name="description"
              rows="4"
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-236 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-238"
              >
                <span>
                  Description
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
      >
        <input
          accept="image/*"
          id="media-photo"
          multiple=""
          style="display:none"
          type="file"
          value=""
        />
        <label
          for="media-photo"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiButton-root MuiButton-contained"
            role="button"
            tabindex="0"
          >
            <span
              class="MuiButton-label"
            >
              <span
                class="MuiButton-startIcon MuiButton-iconSizeMedium"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3 4V1h2v3h3v2H5v3H3V6H0V4h3zm3 6V7h3V4h7l1.83 2H21c1.1 0 2 .9 2 2v12c0 1.1-.9 2-2 2H5c-1.1 0-2-.9-2-2V10h3zm7 9c2.76 0 5-2.24 5-5s-2.24-5-5-5-5 2.24-5 5 2.24 5 5 5zm-3.2-5c0 1.77 1.43 3.2 3.2 3.2s3.2-1.43 3.2-3.2-1.43-3.2-3.2-3.2-3.2 1.43-3.2 3.2z"
                  />
                </svg>
              </span>
              Photos
            </span>
          </span>
        </label>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
      >
        <input
          accept="video/*"
          id="media-video"
          multiple=""
          style="display:none"
          type="file"
          value=""
        />
        <label
          for="media-video"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiButton-root MuiButton-contained"
            role="button"
            tabindex="0"
          >
            <span
              class="MuiButton-label"
            >
              <span
                class="MuiButton-startIcon MuiButton-iconSizeMedium"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19.35 10.04C18.67 6.59 15.64 4 12 4 9.11 4 6.6 5.64 5.35 8.04 2.34 8.36 0 10.91 0 14c0 3.31 2.69 6 6 6h13c2.76 0 5-2.24 5-5 0-2.64-2.05-4.78-4.65-4.96zM14 13v4h-4v-4H7l5-5 5 5h-3z"
                  />
                </svg>
              </span>
              Videos
            </span>
          </span>
        </label>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <p
          class="MuiFormHelperText-root"
          id="my-helper-text"
        >
          Fields marked with (*) are mandatory.
        </p>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        id="btn-next"
        style="margin-top:12px;margin-left:auto"
        tabindex="0"
        type="submit"
      >
        <span
          class="MuiButton-label"
        >
          Next
        </span>
      </button>
    </div>
  </form>,
]
`;
